app/models/piece.rb

```diff
@@ -82,4 +66,21 @@ def move_to!(new_row, new_column)
       end
     end
   end
+
+  def moving_into_check?(new_row, new_column)
+    #We will switch to the new coordinates temporarily and see if we are in check!
+    current_row = self.position_row
+    current_column = self.position_column
+    self.update_attributes(position_row: new_row, position_column: new_column)
+    #Get the status of check?
```

Comment:
I think status comes from the first output and culprit comes from the second output from the game.check? method in lines 96, 94, 99 for the game.rb file. 

http://www.rubyfleebie.com/how-to-return-multiple-values-from-a-method/

The conditional statements seem to be at lines 82 to 86